### PR TITLE
Done

### PR DESCRIPTION
**Pull request by DonatelloTHM**

### CODE REVIEW

File: src/components/SpiceList.js [hunk 1]
```diff
@@ -8,7 +8,23 @@ class SpiceList extends React.Component {
...
+      search: newSearchState
+    })
+  }
+
+  //handles the returned four star checkbox state from the the callback that we sent to the child named Filter. I'm wondering if anyone named their child Filter haha
+  handleCheckedState = (newCheckedState) => {
+    this.setState({
+      fourStarOnly: newCheckedState
+    })
+  }
```
Comment (ihollander) [inline src/components/SpiceList.js:23]: Nice! One trick here you could try for shortening the syntax would be to use [shorthand property names](https://ui.dev/shorthand-properties/#:~:text=With%20Shorthand%20Properties%2C%20whenever%20you,can%20omit%20the%20property%20name.) like this:

```js
handleSearchState = search => {
    this.setState({ search })
  }

  handleCheckedState = fourStarOnly => {
    this.setState({ fourStarOnly })
  }
```

This works as long as the key in state matches the name of the variable you're trying to assign to it.

File: src/components/NewSpice.js [hunk 2]
```diff
@@ -11,21 +11,27 @@ class NewSpice extends React.Component {
   handleSubmit = e => {
     e.preventDefault()
     console.log("submitting...")
+    this.props.newSpiceHandler(this.state)
   }
+  //handles the state for the form inputs
+  handleFormState = evt => {
+    this.setState({[evt.target.name]: evt.target.value})
+  }
```
Comment (ihollander) [inline src/components/NewSpice.js:19]: Nice job making that reusable event handler 💪

**Comment by ihollander** [comment]
Great work! Looks like you've got a solid handle on inverse data flow and passing down callbacks as props.

File: src/components/SpiceList.js (hunk 1 above)
Comment (DonatelloTHM) [reply src/components/SpiceList.js:23]: This is so neat , thank you!

**Pull request closed without merging**